Feat/collabland user verify

msadura opened the pull request:
User verification via collabland endpoint.

Unfortunately, they did not give us a way of creating our own test guild / member so we need to use hardcoded ones to test it:

GuildId 834476259199156245 (space's discordServerId)
UserId 646733536744833054 (discordUserId)

api token: 21hlXT_f0whhXLg9pyXGN.5aMN4Ya64sA7ylxNh2jtp

msadura commented on background/cron.ts:
```diff
@@ -26,6 +27,8 @@ cron.schedule('0 */15 * * * *', proposalTask);
 // Verify token gates and remove users who no longer meet the conditions
 cron.schedule('0 */30 * * * *', verifyTokenGateMembershipsTask);
 
+cron.schedule('0 */3 * * * *', processWebhookMessages);
```
running webhook messages fetching every 3mins - probably value to discuss

mattcasey replied:
Actually, can you change this. Instead of a cron schedule, we should use long polling and then it can just be a function that calls itself recursively. That way we receive messages almost instantaneously: https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/sqs-examples-enable-long-polling.html. It worked well at my last company

mattcasey commented on background/tasks/processWebhookMessages/task.ts:
```diff
@@ -0,0 +1,13 @@
+  log.debug('Running process webhook messages cron job');
+
+  try {
+    await processMessages({ processorFn: processWebhookMessage, maxNumOfMessages: 5 });
```
maxNumOfMessages should be 1, and we should change the queue to FIFO

mattcasey commented on background/tasks/processWebhookMessages/task.ts:
```diff
@@ -0,0 +1,13 @@
+  try {
+    await processMessages({ processorFn: processWebhookMessage, maxNumOfMessages: 5 });
+  } catch (error: any) {
+    log.error(`Error deleting archived pages: ${error.stack || error.message || error}`, { error });
```
error message is from another task

mattcasey commented on lib/webhooks/processWebhookMessage.ts:
```diff
@@ -0,0 +1,60 @@
+        member: { discordId: discordUserId },
+        role
+      } = message?.data as MemberRoleWebhookData;
+      await createAndAssignRolesDiscord({ discordUserId, discordServerId, roles: role });
```
i think you meant to call unassignRolesDiscord() here?

mattcasey commented on lib/webhooks/processWebhookMessage.ts:
```diff
@@ -0,0 +1,60 @@
+        member: { discordId: discordUserId },
+        role
+      } = message?.data as MemberRoleWebhookData;
+      await unassignRolesDiscord({ discordUserId, discordServerId, roles: role });
```
is this supposed to be createAndAssignRolesDiscord()?

mattcasey commented on prisma/schema.prisma:
```diff
@@ -723,6 +724,7 @@ model Role {
   id                        String                     @id @default(uuid()) @db.Uuid
   createdAt                 DateTime                   @default(now())
   createdBy                 String                     @db.Uuid
+  externalId                String?
```
lgtm

mattcasey reviewed (changes requested):
see comments

msadura commented:
Fixed mentioned code issues. I also removed webhook cron and I will work on making FIFO queue + long polling. I think we can deal with them in separate PR, to have discord verification merged
